### DISCUSSION TITLE
Issue with models list

### DISCUSSION THREAD

**Discussion by danstis** [Kilvoctu/aiyabot#117] [Q&A]
Hi, I have a `models.csv` file that looks like this:
```csv
display_name|model_full_name|activator_token
Default||
SD 1.5|external\sd-v1-5-pruned-emaonly.ckpt|
SD 2.1|external\sd-v2-1_768-ema-pruned.ckpt|
Deliberate|external\deliberate_v11.safetensors|
Openjourney|external\mdjrny-v4.ckpt|mdjrny-v4 style
Openjourney v2|external\openjourney-v2.ckpt|
Redshift|external\redshift-diffusion-v1|redshift style
```

When I restart everything (web-ui & aiyabot), then check the data_models list in the draw prompt, I see only one option `Redshift`. If I start typing any of the other options, they do not show up. 
I have also tried doing a `/settings refresh:True`, which shows that is refreshes the global lists, but I still don't see the other options. 

What am I doing wrong here? It was working in the past, but since I pulled the latest commits it seems to have broken.

**Comment by Kilvoctu** [comment]
Heya, can you run the API test on the Web UI and tell me what the response body says?  

It's at http://127.0.0.1:7860/docs#/default/get_sd_models_sdapi_v1_sd_models_get (or wherever your URL is).  
Click `Try it out`, then click on `Execute`

I think in my code, there must be some logic issue when including file extensions in `models.csv`. That block of code is pretty convoluted on finding matches from Web UI.

**Reply by danstis** [reply]
Hey, as requested here is some output from the Swagger UI: 
```json
[
  {
    "title": "external\\deliberate_v11.safetensors [d8691b4d16]",
    "model_name": "external_deliberate_v11",
    "hash": "d8691b4d16",
    "sha256": "d8691b4d16fb59aea959db797350016e24008d59de8d5a960ff7b1cec1fbc11c",
    "filename": "D:\\AI-Art\\models\\external\\deliberate_v11.safetensors",
    "config": null
  },
  {
    "title": "external\\redshift-diffusion-v1.ckpt [ed8c2ee432]",
    "model_name": "external_redshift-diffusion-v1",
    "hash": "ed8c2ee432",
    "sha256": "ed8c2ee43268932a420f5db00b105881770a19c0afd0d35876330e2bbdcce426",
    "filename": "D:\\AI-Art\\models\\external\\redshift-diffusion-v1.ckpt",
    "config": null
  },
  {
    "title": "external\\sd-v1-5-inpainting.ckpt [c6bbc15e32]",
    "model_name": "external_sd-v1-5-inpainting",
    "hash": "c6bbc15e32",
    "sha256": "c6bbc15e3224e6973459ba78de4998b80b50112b0ae5b5c67113d56b4e366b19",
    "filename": "D:\\AI-Art\\models\\external\\sd-v1-5-inpainting.ckpt",
    "config": null
  },
  {
    "title": "external\\sd-v1-5-pruned-emaonly.ckpt [cc6cb27103]",
    "model_name": "external_sd-v1-5-pruned-emaonly",
    "hash": "cc6cb27103",
    "sha256": "cc6cb27103417325ff94f52b7a5d2dde45a7515b25c255d8e396c90014281516",
    "filename": "D:\\AI-Art\\models\\external\\sd-v1-5-pruned-emaonly.ckpt",
    "config": null
  },
  {
    "title": "external\\sd-v2-1_768-ema-pruned.ckpt [ad2a33c361]",
    "model_name": "external_sd-v2-1_768-ema-pruned",
    "hash": "ad2a33c361",
    "sha256": "ad2a33c361c1f593c4a1fb32ea81afce2b5bb7d1983c6b94793a26a3b54b08a0",
    "filename": "D:\\AI-Art\\models\\external\\sd-v2-1_768-ema-pruned.ckpt",
    "config": "D:\\AI-Art\\models\\external\\sd-v2-1_768-ema-pruned.yaml"
  },
  {
    "title": "external\\mdjrny-v4.ckpt [5d5ad06cc2]",
    "model_name": "external_mdjrny-v4",
    "hash": "5d5ad06cc2",
    "sha256": "5d5ad06cc24170b32f25f0180a357e315848000c5f400ffda350e59142fabd68",
    "filename": "D:\\AI-Art\\models\\external\\mdjrny-v4.ckpt",
    "config": null
  },
  {
    "title": "external\\openjourney-v2.ckpt [2fbe5ba6e2]",
    "model_name": "external_openjourney-v2",
    "hash": "2fbe5ba6e2",
    "sha256": "2fbe5ba6e2a842fa99f52500cd600ac33f510b6a76b58874fa92b39c72c5ad57",
    "filename": "D:\\AI-Art\\models\\external\\openjourney-v2.ckpt",
    "config": null
  }
]
```

**Reply by danstis** [reply]
Ahh, to add to this, I checked in `settings.py`, seems that you are matching the entries by their name against the filename or model_name, but both do replacements on `\\` or `/`. My Models.csv has a single `\` as per the dropdown in the UI. 
I figure that I should just replace the backslashes with forward slashes. :)

**Reply by danstis** [reply]
Ok, that fixed it, the Models file should have forward slashes or double back slashes.
If this seems correct, would you like me to update the wiki content? Or raise an issue/PR to add another replacement for a single backslash?

Wiki link: https://github.com/Kilvoctu/aiyabot/wiki/Model-swapping

**Reply by Kilvoctu** [reply]
Ah ok, I see.  
Reviewing just now, I was confused because my models.csv also has single `\`; however, I don't include the file extension. So it seems like a `\\` is needed when including the extension.. It explains why only your Redshift entry worked. 

Well, you saw how my code there looks, lol. My inexperience is showing.  
If you'd like, you can create the PR; otherwise, I'll do it sometime tomorrow my time. Thanks very much for the troubleshooting 👍👍

Afterward, I need to get started on the simple GUI as mentioned at top of #85. I know the models setup has been a headache for people since starting out.

**Reply by danstis** [reply]
Haha, no worries, Can't complain much, your bot is doing awesome in general!
